zipline/gens/tradesimulation.py

```diff
@@ -83,12 +83,11 @@ def perf_key(self):
         return self.EMISSION_TO_PERF_KEY_MAP[
             self.algo.perf_tracker.emission_rate]
 
-    def process_event(self, event):
-        process_trade = self.algo.blotter.process_trade
-        for txn, order in process_trade(event):
-            self.algo.perf_tracker.process_event(txn)
-            self.algo.perf_tracker.process_event(order)
-        self.algo.perf_tracker.process_event(event)
+    def process_event(self, blotter_process_trade, perf_process_event, event):
```

Comment:
because the bound methods are passed in, this should probably become 'private', also the bound methods should probably be private args ('_blotter_process_trade') so that consumers don't mess this up.